From 4ebc2ac4e8ef4452ebe15998077df067ffba6ff6 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 29 Feb 2024 16:58:29 +0200
Subject: [PATCH] [FSW-2066] spi_flash: fix chip_select gpio mask

Due to error in documentation wrong gpio mask was used for spi
flash chip select.

Change-Id: I70ad1ba3829031db7264f6ea9642d2e3ac41d424
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index dd3c035..204579d 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -44,7 +44,7 @@
 
 unsigned int get_cs_pad_sel_flash(void)
 {
-	return mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE + mmPSOC_PAD_CONF_PAD_SEL_18;
+	return mmIO_0_CFG_HALF_0_PSOC_PSOC_CFG_PAD_CONF_BASE + mmPSOC_PAD_CONF_PAD_SEL_8;
 }
 
 void spi_flash_base_get(unsigned int *regs)
@@ -55,7 +55,7 @@
 void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
 		     unsigned int *out_en_offset)
 {
-	*mask = BIT(18);
+	*mask = BIT(8);
 	*data_offset = mmIO_0_CFG_HALF_0_PSOC_GPIO0_BASE + mmGPIO_DATAOUT;
 	*out_en_offset = mmIO_0_CFG_HALF_0_PSOC_GPIO0_BASE + mmGPIO_OUTENSET;
 }
